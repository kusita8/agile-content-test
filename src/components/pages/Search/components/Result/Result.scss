.result {
  display: flex;
  flex-direction: column;
  gap: 0.6rem;

  &:not(.result--active-display) {
    .result__description {
      overflow: hidden;
      text-overflow: ellipsis;
      display: -webkit-box;
      -webkit-line-clamp: 3;
      -webkit-box-orient: vertical;
    }
  }

  &__url {
    font-size: 1.4rem;
    color: var(--grey-b);
  }

  &__link {
    text-align: left;
    h4 {
      font-size: 2rem;
      color: var(--blue-a);
      font-weight: 500;
      margin-top: 0.6rem;
    }
  }

  &__description {
    font-size: 1.4rem;
    color: var(--grey-b);
  }
}
