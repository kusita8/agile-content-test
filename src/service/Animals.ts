import { faker } from "@faker-js/faker";

const results = [...new Array(100)].map(() => {
  const type = faker.animal.type() as keyof typeof faker.animal;

  return {
    type,
    id: faker.datatype.uuid(),
    url: faker.internet.url(),
    title: faker.animal[type](),
    description: faker.lorem.sentences(),
    image: faker.image.animals(644, 362, true),
  };
});

export const getAnimals = async (params: Record<string, string>) => {
  // fake delay
  await new Promise((res) => setTimeout(res, 1000));

  const { query } = params;
  const lowercaseQuery = query.toLowerCase();

  return results.filter(
    (animal) =>
      lowercaseQuery &&
      (animal.type.toLowerCase().includes(lowercaseQuery) ||
        animal.title.toLowerCase().includes(lowercaseQuery))
  );
};
