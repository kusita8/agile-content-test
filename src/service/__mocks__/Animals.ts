import { faker } from "@faker-js/faker";

export const getAnimal = () => {
  const type = faker.animal.type() as keyof typeof faker.animal;

  return {
    type: faker.animal.type(),
    id: faker.datatype.uuid(),
    url: faker.internet.url(),
    title: faker.animal[type](),
    description: faker.lorem.sentences(),
    image: faker.image.animals(644, 362, true),
  };
};

export const getAnimals = async (params: Record<string, string>) => {
  const { query } = params;

  const type = faker.animal.type() as keyof typeof faker.animal;

  return [...new Array(2)]
    .map(() => ({
      type: faker.animal.type(),
      id: faker.datatype.uuid(),
      url: faker.internet.url(),
      title: faker.animal[type](),
      description: faker.lorem.sentences(),
      image: faker.image.animals(644, 362, true),
    }))
    .filter(
      (animal) =>
        query &&
        animal.type.toLocaleLowerCase().includes(query.toLocaleLowerCase())
    );
};
