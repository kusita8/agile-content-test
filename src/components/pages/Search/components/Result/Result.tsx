import { FC } from "react";

import { IAnimal } from "src/types/Animal.interface";
import { getClassName } from "src/utils/getClassName";

import "./Result.scss";

interface ResullProps extends IAnimal {
  onClick?: () => void;
  activeDisplay?: boolean;
}

export const Result: FC<ResullProps> = ({
  url,
  title,
  description,
  activeDisplay,
  onClick,
}) => {
  const linkContent = (
    <>
      <div className="result__url">{url}</div>
      <h4>{title}</h4>
    </>
  );

  return (
    <div
      className={getClassName(
        "result",
        activeDisplay && "result--active-display"
      )}
    >
      {activeDisplay ? (
        <a
          className="result__link"
          href={url}
          rel="noopener noreferrer"
          target="_blank"
        >
          {linkContent}
        </a>
      ) : (
        <button className="result__link" onClick={onClick}>
          {linkContent}
        </button>
      )}
      <div className="result__description">{description}</div>
    </div>
  );
};
