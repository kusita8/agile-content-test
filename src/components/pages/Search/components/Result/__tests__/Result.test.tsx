/* eslint-disable no-restricted-imports */
import { Result } from "src/components/pages/Search/components/Result/Result";
import { getAnimal } from "src/service/__mocks__/Animals";
import { renderWithProviders } from "src/tests/renderWithProviders";

describe("Result", () => {
  test("it should match the snapshot", () => {
    const { asFragment } = renderWithProviders(<Result {...getAnimal()} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
