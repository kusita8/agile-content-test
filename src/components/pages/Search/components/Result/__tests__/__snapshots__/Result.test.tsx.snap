// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Result > it should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <div
      class="header"
    >
      <p>
        <b>
          Agile content
        </b>
         Frontend test
      </p>
      <div
        class="header__right"
      >
        <svg
          class="gb_p"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,8c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM12,20c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM6,20c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM6,14c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM12,14c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM16,6c0,1.1 0.9,2 2,2s2,-0.9 2,-2 -0.9,-2 -2,-2 -2,0.9 -2,2zM12,8c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM18,14c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM18,20c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2z"
          />
        </svg>
        <div
          class="header__user"
        >
          <img
            alt="Profile image"
            src="https://images.unsplash.com/photo-1511367461989-f85a21fda167?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2231&q=80"
          />
        </div>
      </div>
    </div>
    <div
      class="result"
    >
      <button
        class="result__link"
      >
        <div
          class="result__url"
        >
          http://voluminous-ego.com
        </div>
        <h4>
          Japanese common catfish
        </h4>
      </button>
      <div
        class="result__description"
      >
        Cumque fugiat accusamus. Delectus itaque accusantium. Sunt quas quod reprehenderit corrupti cumque necessitatibus porro ad amet. Delectus eveniet quibusdam. Ratione explicabo facere voluptatum perspiciatis blanditiis.
      </div>
    </div>
    <div
      class="footer"
    >
      <p>
        Â© Google 2023
      </p>
      <p>
        version: 0.1.0
      </p>
    </div>
  </div>
</DocumentFragment>
`;
